File: branches/project/iOS_Console_2_1_0_Betas/Classes/Switch.m
/*
 * OpenRemote, the Home of the Digital Home.
 * Copyright 2008-2011, OpenRemote Inc.
 *
 * See the contributors.txt file in the distribution for a
 * full listing of individual contributors.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
#import "Switch.h"
#import "Image.h"
#import "SensorState.h"

@implementation Switch

@synthesize onImage, offImage;

//get element name, must be overriden in subclass
- (NSString *) elementName
{
	return SWITCH;
}

#pragma mark Delegate methods of NSXMLParser  


- (id)initWithXMLParser:(NSXMLParser *)parser elementName:(NSString *)elementName attributes:(NSDictionary *)attributeDict parentDelegate:(NSObject<NSXMLParserDelegate> *)parent
{
    self = [super init];
	if (self) {
		componentId = [[attributeDict objectForKey:ID] intValue];
		
		xmlParserParentDelegate = [parent retain];
		[parser setDelegate:self];
	}
	return self;
}

/**
 * Fill the switch on/off state images .
 */
- (void)parser:(NSXMLParser *)parser didEndElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qName
{
	if ([elementName isEqualToString:[self elementName]]) {	
		
		for (SensorState *state in sensor.states) {
			Image *img = [[Image alloc] init];
			img.src = [state.value copy];
			if ([[state.name lowercaseString] isEqualToString:ON]) {
				onImage = [img retain];
			} else if ([[state.name lowercaseString] isEqualToString:OFF]) {
				offImage = [img retain];
			}
			[img release];
		}
				
 		[parser setDelegate:xmlParserParentDelegate];
		[xmlParserParentDelegate release];
		xmlParserParentDelegate = nil;
	}
}

- (void)dealloc
{
	[onImage release];
	[offImage release];
	[xmlParserParentDelegate release];
	[super dealloc];
}


@end

File: workspace/juha/oldstuff/Controller_EP_SNAP_20111115/src/org/openremote/controller/model/event/Switch.java
/*
 * OpenRemote, the Home of the Digital Home.
 * Copyright 2008-2011, OpenRemote Inc.
 *
 * See the contributors.txt file in the distribution for a
 * full listing of individual contributors.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package org.openremote.controller.model.event;

import org.openremote.controller.protocol.Event;

/**
 * TODO
 *
 * @author <a href="mailto:juha@openremote.org">Juha Lindfors</a>
 */
public class Switch extends Event<String>
{
  public enum State { ON, OFF }

  private String value;
  private State eventState;

  public Switch(int sourceID, String sourceName, String value, State originalState)
  {
    super(sourceID, sourceName);

    this.eventState = originalState;
    this.value = value;
  }


  public Switch.State getOriginalState()
  {
    return eventState;
  }

  @Override public String getValue()
  {
    return value;
  }

  @Override public void setValue(String value)
  {
    this.value = value;
  }

  @Override public String serialize()
  {
    //return eventState.toString().toLowerCase();
    return getValue();
  }

  @Override public boolean isEqual(Object o)
  {
    if (o == null)
    {
      return false;
    }

    if (o == this)
    {
      return true;
    }

    if (o.getClass() != this.getClass())
    {
      return false;
    }

    Switch s = (Switch)o;

    return s.getSourceID().equals(this.getSourceID())
        && s.getSource().equals(this.getSource())
        && s.getValue().equals(this.getValue());
  }

}

